lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -38,8 +39,11 @@ def initialize(serializer, options)
         end
 
         def as_json
```

Comment:
arguably, if `serializer.object` is nil, the return value should be `nil`, not `{ type: type}`, and consider it a 'nil' identifier